### PR TITLE
[Dataset] RedditDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Description
#1776

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/reddit.py [hunk 1]
```diff
@@ -1,65 +1,236 @@
...
+
+    @property
+    def num_classes(self):
+        r"""Number of classes for each node."""
+        return 41
+
+    @property
+    def num_labels(self):
+        deprecate_property('dataset.num_labels', 'dataset.num_classes')
+        return 41
```
Comment (classicsong) [inline python/dgl/data/reddit.py:181]: You can return self.num_classes

File: python/dgl/data/reddit.py [hunk 2]
```diff
@@ -1,65 +1,236 @@
...
+            >>> labels = graph.ndata['label']
+
+    This is a graph dataset from Reddit posts made in the month of September, 2014.
+    The node label in this case is the community, or “subreddit”, that a post belongs to.
+    The authors sampled 50 large communities and built a post-to-post graph, connecting
+    posts if the same user comments on both. In total this dataset contains 232,965
+    posts with an average degree of 492. We use the first 20 days for training and the
+    remaining days for testing (with 30% used for validation).
+
+    Reference: http://snap.stanford.edu/graphsage/
```
Comment (hetong007) [inline python/dgl/data/reddit.py:52]: Wrap with rst hyperlink format:

```
Reference:  `GraphSAGE<http://snap.stanford.edu/graphsage/>`_.
```

Comment (hetong007) [reply python/dgl/data/reddit.py:52]: I saw other datasets are also using plain text format. Probably leave it.

Comment (HuXiangkun) [reply python/dgl/data/reddit.py:52]: We'll sync about the doc string format.

**Pull request merged**